Added proxy support

cesarpinera opened the pull request:
I'm adding proxy support to the creation of the DynamoDB client. 

In the credential map two new keys :proxy-host and :proxy-port can be specified. If they exist, a ClientConfiguration instance is added to the constructor of the AmazonDynamoDB object. 

E.g.:

{:access-key "AAA" :secret-key "ZZZ" :proxy-host "proxy-us.intel.com" :proxy-port 911}

Commit:
Addded proxy configuration to the credential map and to the creation of the dynamodb client

Commit:
Addded proxy configuration to the credential map and to the creation of the dynamodb client

weavejester commented:
You could write it a little more compactly like:

``` clojure
(defn- db-client*
  "Get a AmazonDynamoDBClient instance for the supplied credentials."
  [{:keys [access-key secret-key endpoint proxy-host proxy-port]}]
  (let [aws-creds     (BasicAWSCredentials. access-key secret-key)
        client-config (ClientConfiguration.)]
    (when proxy-host (.setProxyHost client-config proxy-host))
    (when proxy-port (.setProxyPort client-config proxy-port))
    (let [client (AmazonDynamoDBClient. aws-creds client-config)]
      (when endpoint (.setEndpoint client endpoint))
      client)))
```

Since:

``` clojure
(AmazonDynamoDBClient. aws-creds)
```

Is equivalent to:

``` clojure
(AmazonDynamoDBClient. aws-creds (ClientConfiguration.))
```

Commit:
Added proxy support

Added proxy-host and proxy-port keys to the credential map and to the
db-client* fn.

cesarpinera commented:
Excellent suggestion. I made the changes. I do believe it's important to test for both proxy-host and proxy-port, otherwise ClientConfiguration will be incomplete if one of the values is missing.

weavejester commented:
Why is it a bad thing if ClientConfiguration is incomplete? It's generally better to fail noisily, and I'd prefer an exception over a silent failure.

Commit:
Proxy-host and Proxy-port are now set independently of each other.

cesarpinera commented:
Got it. I made proxy-host and proxy-port independent of each other.

weavejester commented:
Could you squash your commits? Then I'll merge.

Commit:
Merge branch 'proxy-support' of https://github.com/cesarpinera/rotary into proxy-support

cesarpinera commented:
It took me a litte while to find out how to squash commits that have already been pushed. I hope I did it right. Let me know.

weavejester commented:
I'm afraid you haven't done it right. Run the following command:

```
git rebase -i master
```

This will open up an editor with an interactive rebase. Use the editor to change all but one commit to be squashed. Once you've done that, you should end up with just one commit ahead of master. You can use `git log` to check.

In order to push the commits, run:

```
git push -f
```

The `-f` part will force the push to overwrite the old.

cesarpinera commented:
I'm sorry but I cannot seem to find a way to do that. It appears like the commits have already been squashed and pushed to the branch. 

If you prefer, I can create a new branch, put a single commit there and then create a new pull request from it. 

Thank you for being this patient.
